lib/features/digital_guide_view/tabs/adapted_toilets/presentation/adapted_toilet_detail_view.dart

```diff
@@ -0,0 +1,152 @@
+import "package:auto_route/auto_route.dart";
+import "package:flutter/material.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+
+import "../../../../../config/ui_config.dart";
+import "../../../../../theme/app_theme.dart";
+import "../../../../../utils/context_extensions.dart";
+import "../../../../../widgets/detail_views/detail_view_app_bar.dart";
+import "../../../../../widgets/my_cached_image.dart";
+import "../../../../../widgets/my_point_widget.dart";
+import "../../../presentation/widgets/digital_guide_nav_link.dart";
+import "../data/models/adapted_toilet.dart";
+
+@RoutePage()
+class AdaptedToiletDetailView extends ConsumerWidget {
+  const AdaptedToiletDetailView({
+    required this.adaptedToilet,
+  });
+
+  final AdaptedToilet adaptedToilet;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return Scaffold(
+      appBar: DetailViewAppBar(
+        actions: const [],
+      ),
+      body: Padding(
+        padding: const EdgeInsets.symmetric(
+          horizontal: DigitalGuideConfig.heightBig,
+        ),
+        child: CustomScrollView(
+          slivers: [
+            SliverList(
+              delegate: SliverChildListDelegate([
+                Text(
+                  () {
+                    switch (adaptedToilet.hasAdditionalPurpose) {
+                      case 1:
+                        return context.localize.adapted_toilet_description;
+                      case 2:
+                        return context.localize.adapted_toilet_description_men;
+                      case 3:
+                        return context
+                            .localize.adapted_toilet_description_women;
+                      default:
+                        return context.localize.adapted_toilet_description;
```

Comment:
You can do it in extension and I think it's better to use switch expression not switch statement in this case.